src/triage/component/catwalk/storage.py

```diff
@@ -652,3 +658,9 @@ class TrainMatrixType(object):
     evaluation_obj = TrainEvaluation
     prediction_obj = TrainPrediction
     is_test = False
+
+
+class ProductionMatrixType(object):
+    string_name = "production"
+    prediction_obj = ListPrediction
+
```

Comment:
@tweddielin

Reply:
I think `is_test` is only used in `ModelEvaluator` to decide if it should use `testing_metric_groups` or `training_metric_groups`. For `ProductionMatrixType`, at least for generating risk list, there's no evaluation involved. I'm not sure why it needs `is_test` regardless of good or poor naming.

Reply:
Hmmm... would it make sense to remove the `is_test` boolean entirely and simply check if `matrix_type.string_name == 'test'` in evaluation.py? As it is, it seems a little redundant with the string name, especially if it doesn't fit well with the production matrix type here.

Reply:
I'm not fond of comparisons against magic strings. What about making it more specifically about the metric groups?  Like what if the field was metric_config_key, and for train it was 'training_metric_groups', for test it was 'testing_metric_groups', and for production it was None?

Reply:
Yeah, that's a fair point about avoiding magic strings -- having it keep track of the appropriate `metric_groups` types it needs to use makes sense to me if that's the only place `is_test` is getting used downstream regardless.

Reply:
Sorry, just coming into this discussion quite late. It seems to me that which metrics to use is not a property of the matrix but of the task being performed, and the discussion here is exposing decisions we committed to early on and haven't tried to address.

I feel like we always saw the `matrix_type` as a bad pattern but implemented it for expediency to flag (poorly) the different behavior of label imputation in train/test for the inspection problem type. Is my memory accurate? There were conceptual issues with this all along because matrix_type was never a good name and it prevented us from reusing matrices from testing for training in the EWS problem type.

It also seems like which metrics to use is not properly a property of the matrix but of the task being performed. Perhaps we should be passing the metrics groups to the evaluator instead of encoding in the matrix whether it should use train or test metrics.

I think maybe we can move forward with this as a way to get the feature we want now, but this discussion is making the illogic of these matrix attributes more apparent and we may have some deeper technical debt to service here rather than trying to make the current situation totally coherent.

Reply:
I don't know if you all discussed this on Tuesday, but here are a couple of possible solutions for Triage v 5.0.0. The first is probably easier to implement, but I think it is the less good solution:

- Eliminate both the `matrix_type` and `is_test` properties of matrices
- Include missing labels as missing in all matrices
- Replace `include_missing_labels_in_train_as` with two keys: `impute_missing_labels_at_train` and `impute_missing_labels_at_test`. These can be `True` or `False`. If `True`, you must also specify the value. We already have lots of duplicated keys for train/test behavior, so this fits an existing configuration pattern. These can be set to defaults when there are inspection and EWS entry points
- The Architect writes matrices without label imputation, allowing them to be reused for any purpose
- Catwalk does the label imputation on the fly when it is training or testing and always writes the labels to the database. This has high storage costs, but the reason we had to decide on the label imputation at matrix-building time initially was because we had not yet conceived of the `train_results` schema. Now that Triage can write train predictions, it _can_ write train labels to the database, and the `matrix_type` concept is no longer needed.
- The ModelEvaluator takes only a single set of metrics, and Catwalk creates a separate ModelEvaluator object for training and testing.
- Separate matrix storage, rather than separate matrix types, is introduced for Production and Development

An alternative:

- Eliminate both the `matrix_type` and `is_test` properties of matrices
- Include missing labels as missing in all matrices
- Replace `include_missing_labels_in_train_as` with two keys: `impute_missing_labels_at_train` and `impute_missing_labels_at_test`. These can be `True` or `False`. If `True`, you must also specify the value. We already have lots of duplicated keys for train/test behavior, so this fits an existing configuration pattern. These can be set to defaults when there are inspection and EWS entry points
- Instead of storing labels in the train/test matrices, we begin storing the design matrices and the labels separately. Train labels and test labels have separate stores, and the imputation behavior is a key in the label metadata.
- The `replace` flag checks the imputation flag on the labels to determine whether it needs to create new labels
- Design matrices are combined with the correct labels on reading, and rows are dropped if there is no label and the imputation key is `False`
- You can still skip writing train results to conserve db storage
- Production/predict-forward Uses design matrices but not labels

Reply:
Moved the above comment to #855